docs/source/external_devices/payload_handling_area.rst

```diff
@@ -0,0 +1,284 @@
+   :header-rows: 1
+
+   * - Topic Name
+     - Msg Type
+     - Note
+   * -  **SUBSCRIBE**
+     - 
+     - 
+   * - /set_mode
+     - `example_interfaces/msg/Int32 <https://docs.ros2.org/foxy/api/example_interfaces/msg/Int32.html>`_
+     - Sets the mode: 0 for manual, 1 for auto.
+   * - /set_duration
+     - `example_interfaces/msg/Float32 <https://docs.ros2.org/foxy/api/example_interfaces/msg/Float32.html>`_
+     - Sets the duration for auto-triggering actions.
+   * - /trigger
+     - `example_interfaces/msg/Empty <https://docs.ros2.org/foxy/api/example_interfaces/msg/Empty.html>`_
+     - Manually triggers an action.
+
+Sink Area
+---------
+
+`BP_SinkArea` is a child class of `BP_PayloadHandlerBase`.
+
+- `/trigger` action 
+    deletes the actor.
+
+- **Condition for Auto Mode**: 
+    The action is triggered when an actor overlaps with the area.
+
+
+.. video:: ../_static/videos/sink_area.mp4
+    :width: 750
+    :height: 450
+
+*Video: Sink Area*
+
+
+Source Area
+-----------
+
+`BP_SourceArea` is a child class of `BP_PayloadHandlerBase`. The source area contains matrix spots where actors are spawned and attached to the area.
+
+- `/trigger` action:
+   spawns and attaches an actor to the spots.
+
+- **Condition for Auto Mode**: 
+    The action is triggered when the actor in the spot is detached from the area.
+
+Depending on the `/source_mode`, the `/trigger` action will either spawn one actor at a time or all actors when all spots are empty.
+
+.. video:: ../_static/videos/source_area.mp4
+    :width: 750
+    :height: 450
+
+*Video: Source Area*
+
+Parameters for Source Area
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+.. list-table::
+   :header-rows: 1
+
+   * - Param Name
+     - Type (Default)
+     - Note
+   * -  **ROS JSON SPAWN PARAMETER**
+     - 
+     - 
+   * - /source_mode
+     - int32 (0)
+     - 
+        1. Spawns one actor at a time.  
+        2. Spawns all actors together when all spots are empty.
+   * - /num
+     - vector (x:1, y:1, z:1)
+     - Dimensions of the source spots.
+   * - /clearance
+     - vector (x:1, y:1, z:1)
+     - Clearance between spots.
+   * - /spawn_actor_classes
+     - string[]
+     - List of actor classes spawned by the area.
+
+ROS 2 API for Source Area
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. list-table::
+   :header-rows: 1
+
+   * - Topic Name
+     - Msg Type
+     - Note
+   * -  **SUBSCRIBE**
+     - 
+     - 
+   * - /set_spawn_mode
+     - `example_interfaces/msg/Int32 <https://docs.ros2.org/foxy/api/example_interfaces/msg/Int32.html>`_
+     - Sets the spawn mode:  
+
+       0. Spawns one actor at a time,  
+       1. Spawns all actors together.
+
+Teleport Area and Manager
+-------------------------
+
+`BP_TeleportArea` is a child class of `BP_SinkArea`, and `BP_TeleportAreaManager` is a child class of `BP_PayloadHandlerBase`. `BP_TeleportAreaManager` holds a list of `BP_TeleportArea` instances, and manually triggering its action will trigger all teleport areas in the list.
```

Comment:
BP_TeleportArea being a child of BP_SinkArea, does it means the object is deleted then spawned?

Reply:
no. just use same logic to trigger event